non-protocol-specs/0011-NP-CLIE-client-interaction-tests.md

```diff
@@ -7,25 +7,28 @@ This specification contains a set of tests/acceptance criteria that clients (wal
-1. Submit one delegation less than (`SpamProtectionMaxDelegations` = 390) , i.e., 389 delegations in the same epoch from an account that has more than `SpamProtectionMinTokensForDelegation` tokens). Once all delegations are processed, submit 100 new delegations at once (still in the same epoch).  Verify that the wallet does not allow more than 1 delegation to be submitted simultaneously, i.e., cause the account to get blocked. Submit 100 new delegations at once in the next epoch and verify that the wallet allows it (and the account does not get blocked).(<a name="0011-NP-CLIE-015" href="#0011-NP-CLIE-015">0011-NP-CLIE-015</a>)
-1. Create a key with sufficient tokens to be allowed to submit proposals and delegations (i.e., more than `SpamProtectionMinTokensForDelegation` and `SpamProtectionMinTokensForProposal`. Submit random (but valid) votes, delegation changes, transactions and proposals in sets of at least 13 each at the same time, at a rate of one set per second. Let that run for 4 epochs, with an epoch length of at least 100 blocks. Verify that the account never gets banned, and that the client never crashed. (<a name="0011-NP-CLIE-016" href="#0011-NP-CLIE-016">0011-NP-CLIE-016</a>)
-1. Create 20 keys with sufficient tokens to be allowed proposals and delegations (i.e., more than `SpamProtectionMinTokensForDelegation` and `SpamProtectionMinTokensForProposal`. For all keys (within their respective wallets), submit random (but valid) votes, delegation changes, transactions and proposals in sets of at least 13 each at the same time, at a rate of one set per second. Let that run for 4 epochs, with an epoch length of at least 100 blocks. Verify that no account ever gets banned, and that the clients never crash. (<a name="0011-NP-CLIE-017" href="#0011-NP-CLIE-017">0011-NP-CLIE-017</a>)
+    - This is communicated to the wallet, and wallet uses the new parameter for each transaction tied to a block with a height higher than the one in which the change happened. The wallet should submit no more than `spam.pow.numberofTxPerBlock` number of transactions when tied to a block higher than the one in which the change happened. In case of excess transactions (i.e. greater than `spam.pow.numberofTxPerBlock`), transactions should not be sent and returns an appropriate error message OR delays sending the transactions. (<a name="0011-NP-CLIE-008" href="#0011-NP-CLIE-008">0011-NP-CLIE-008</a>)
```

Comment:
I'd replace 'delays the transaction' with 'reprocesses the transaction (which may involve a new PoW and potentially delaying it)'. The reason is that it still may be possible to submit the transaction immediately if another past block is used for the spam protection, which would not be covered by the original formulation

Reply:
Amendments made as suggested above.